Core/AppRuntime/Source/TaskChain.cpp

```diff
@@ -0,0 +1,69 @@
+#include "TaskChain.h"
+
+namespace Babylon
+{
+    TaskChain::TaskChain(std::function<void()> threadProcedure)
```

Comment:
It seems weird that this has to be passed in, and the assumption is that this function eventually calls `RunTaskChain`. Why can't `TaskChain` start it's own work queue thread, and the `TaskChain` consumer just call `Append` to get work injected into the thread?

Reply:
It might be possible to do it that way; I'll have to think about it more. This is a fairly direct port of the existing mechanism, and the main reasoning behind that mechanism is because of stack variables. In particular, V8 wants its initialization and state to be stored on the stack, and it goes to great lengths to make it difficult to do anything else. Thus, to use V8 correctly, we need to actually call `TaskChain::RunTaskChain` from inside the scope that contains the V8 state, rather than calling a task at the start to set up V8 and presumably calling another later to tear it down.